app/controllers/posts_controller.rb

```diff
@@ -1,5 +1,14 @@
 class PostsController < ApplicationController
   before_action :authenticate_user!, only: %i[new create]
+
+  def destroy
+    @post = Post.find(params[:id])
+    authorize! :destroy, @post
+    author = @post.author
+    @post.destroy
+    redirect_to user_posts_path(author), notice: 'Post was successfully deleted.'
```

Comment:
- [ ] as part of the requirements , you need to use **CanCanCan** for Post authorization in `Post Controller` by adding   load_and_authorize_resource , kindly fix it 
```ruby
 class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end
```